### PR TITLE
Wait until add-on settings window is closed before running the daemon

### PR DESCRIPTION
**Pull request by elgatito**
Currently we face multiple daemon restart tries when it's a new installation. 
This change checks for addon settings being active, and wait till it's closed, before starting the daemon.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Translation

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: resources/site-packages/quasar/daemon.py [hunk 1]
```diff
@@ -181,6 +181,16 @@ def start_quasard(**kwargs):
         kwargs["env"] = env
         kwargs["close_fds"] = True
 
+    wait_counter = 0
+    while xbmc.getCondVisibility('Window.IsVisible(10140)'):
+        log.debug('Addon Settings opened, waiting before start')
```
Comment (scakemyer) [inline resources/site-packages/quasar/daemon.py]: Over-capitalization ;) Should be `Add-on settings opened, waiting before starting`

File: resources/site-packages/quasar/daemon.py [hunk 2]
```diff
@@ -181,6 +181,16 @@ def start_quasard(**kwargs):
...
+        xbmc.sleep(1000)
+        wait_counter += 1
+
+        # Too much to wait, try to run it
+        if wait_counter > 60:
```
Comment (scakemyer) [inline resources/site-packages/quasar/daemon.py]: 60 seconds sounds too short to wait for the user to choose initial settings. You can remove the comment too, what this does is obvious enough.

File: resources/site-packages/quasar/daemon.py [hunk 3]
```diff
@@ -181,6 +181,16 @@ def start_quasard(**kwargs):
         kwargs["env"] = env
         kwargs["close_fds"] = True
 
+    wait_counter = 0
+    while xbmc.getCondVisibility('Window.IsVisible(10140)'):
+        log.debug('Addon Settings opened, waiting before start')
+        xbmc.sleep(1000)
```
Comment (scakemyer) [inline resources/site-packages/quasar/daemon.py]: Go with `time.sleep(1)` instead.

**Comment by elgatito** [comment]
Commit updated.
Now waiting for max 256 seconds.

File: resources/site-packages/quasar/daemon.py [hunk 4]
```diff
@@ -181,6 +181,15 @@ def start_quasard(**kwargs):
         kwargs["env"] = env
         kwargs["close_fds"] = True
 
+    wait_counter = 1
+    while xbmc.getCondVisibility('Window.IsVisible(10140)'):
+        if wait_counter > 256:
```
Comment (scakemyer) [inline resources/site-packages/quasar/daemon.py]: > We're not programs Gerty, we're people

Go with 300, as in 5 minutes, just increase the counter by 1, and `time.sleep(1)`
256 seconds means nothing to humans...

File: resources/site-packages/quasar/daemon.py [hunk 5]
```diff
@@ -181,6 +181,15 @@ def start_quasard(**kwargs):
...
+            break
+
+        log.debug('Addon settings opened, waiting before start')
```
Comment (scakemyer) [inline resources/site-packages/quasar/daemon.py]: `log.info('Add-on settings currently opened, waiting before starting...')`

File: resources/site-packages/quasar/daemon.py [hunk 6]
```diff
@@ -181,6 +181,14 @@ def start_quasard(**kwargs):
         kwargs["env"] = env
         kwargs["close_fds"] = True
 
+    wait_counter = 1
+    while xbmc.getCondVisibility('Window.IsVisible(10140)'):
+        if wait_counter > 300:
+            break
+
```
Comment (scakemyer) [inline resources/site-packages/quasar/daemon.py]: Last tiny nitpick: removing that blank line

**Comment by scakemyer** [comment]
As we talked about though, this doesn't actually solve the add-on instantly launching twice on first install though...

**Comment by elgatito** [comment]
Yup. This change solves situation when Quasar gets into cyclic restarts, when required options are not set and it opens settings window but does not give time to set anything.

**Pull request merged**